src/frequenz/channels/_broadcast.py

```diff
@@ -71,12 +71,13 @@ async def recv(id: int, receiver: channel.Receiver) -> None:
         Check the `tests` and `benchmarks` directories for more examples.
     """
 
-    def __init__(self, name: str, *, resend_latest: bool = False) -> None:
+    def __init__(self, *, name: str | None, resend_latest: bool = False) -> None:
```

Comment:
I'm not so sure about making the name parameter of `Broadcast` channels optional, because people would just create channels without thinking of names, and we'll have logs complaining about random channels filling up, and it becomes super hard to find out where the bottle neck is.

Reply:
Not really strong reasons, but no need for an extra `import` and `Optional` was just a library solution until `|` got properly implemented in Python 3.10.

Reply:
> I'm not so sure about making the name parameter of `Broadcast` channels optional, because people would just create channels without thinking of names, and we'll have logs complaining about random channels filling up, and it becomes super hard to find out where the bottle neck is.

Mmm, I see. For me it was weird to require an argument that it is only used for logging, but I can see how otherwise people might need to learn the hard way to use custom names.

Also my feeling is, this is only a special case now in `Broadcast` because is what we use the most, but following that argument, `Anycast` should probably also log when a sender is being blocked because the buffer is full, as it is also a sign of receivers not being able to keep up.

Also, this might apply to other classes too, like `Merge`or whatever, where you could potentially have issues with one instance and that get logged and then you don't know which one is giving issues (although I know this is more a theoretical issue). But since this is an interface issue, and it is hard to know what future requirements would be, if names are really important we might think about making them required even if they are mostly unused for now (except for the `str`/`repr` implementation).

I know the above might be a bit extreme, but at least for channel classes, I think we should have a consistent approach and have all names optional or required.

Reply:
Yes, I agree on consistency, and we need it to be mandatory only for the channels, because everything else can derive the names for logging from the channel.

Reply:
OK, I will make it mandatory for all channels then.

Reply:
Done